catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/FederationAdminServiceImpl.java

```diff
@@ -66,147 +131,867 @@ public FederationAdminServiceImpl() {
+        Map<String, Serializable> properties = new HashMap<>();
+        Subject systemSubject = security.getSystemSubject();
+        properties.put(SecurityConstants.SECURITY_SUBJECT, systemSubject);
+
+        DeleteRequest deleteRequest = new DeleteRequestImpl(serializableIds,
+                RegistryObjectMetacardType.REGISTRY_ID,
+                properties,
+                destinations);
+        try {
+            catalogFramework.delete(deleteRequest);
+        } catch (IngestException | SourceUnavailableException e) {
+            String message = "Error deleting registry entries by registry id.";
+            LOGGER.error("{} Registry Ids provided: {}", message, registryIds);
+            throw new FederationAdminException(message, e);
+        }
     }
 
     @Override
     public void deleteRegistryEntriesByMetacardIds(List<String> metacardIds)
             throws FederationAdminException {
-
+        deleteRegistryEntriesByMetacardIds(metacardIds, null);
     }
 
     @Override
     public void deleteRegistryEntriesByMetacardIds(List<String> metacardIds,
             Set<String> destinations) throws FederationAdminException {
+        if (CollectionUtils.isEmpty(metacardIds)) {
+            throw new FederationAdminException(
+                    "An empty list of metacard ids to be deleted was received. Nothing to delete.");
+        }
+
+        List<Serializable> serializableIds = new ArrayList<>(metacardIds);
+        Map<String, Serializable> properties = new HashMap<>();
+        Subject systemSubject = security.getSystemSubject();
+        properties.put(SecurityConstants.SECURITY_SUBJECT, systemSubject);
+
+        DeleteRequest deleteRequest = new DeleteRequestImpl(serializableIds,
+                Metacard.ID,
+                properties,
+                destinations);
+        try {
+            catalogFramework.delete(deleteRequest);
+        } catch (IngestException | SourceUnavailableException e) {
+            String message = "Error deleting registry entries by metacard ids.";
+            LOGGER.error("{} Metacard Ids provided: {}", message, metacardIds);
+            throw new FederationAdminException(message, e);
+        }
+    }
+
+    @Override
+    public List<Metacard> getRegistryMetacards() throws FederationAdminException {
+
+        Filter filter =
+                FILTER_FACTORY.and(FILTER_FACTORY.like(FILTER_FACTORY.property(Metacard.CONTENT_TYPE),
+                        RegistryConstants.REGISTRY_NODE_METACARD_TYPE_NAME),
+                        FILTER_FACTORY.like(FILTER_FACTORY.property(Metacard.TAGS),
+                                RegistryConstants.REGISTRY_TAG));
 
+        return getRegistryMetacardsByFilter(filter);
     }
 
     @Override
-    public void updateRegistryEntry(Metacard metacard) throws FederationAdminException {
+    public List<Metacard> getLocalRegistryMetacards() throws FederationAdminException {
 
+        List<Filter> filters = new ArrayList<>();
```

Comment:
The same basic filters are being reused many times.  Might see if there is a way to extract the similar parts into a method.